File: src/main/java/com/example/app/service/AuthService.java
package com.example.app.service;

import com.example.app.utils.exceptions.*;
import com.example.app.utils.model.JwtResponse;
import com.example.app.utils.model.entities.User;
import jakarta.mail.MessagingException;
import lombok.NonNull;
import org.springframework.stereotype.Service;


@Service
public interface AuthService {

    JwtResponse login(@NonNull User user) throws UserNotFoundException, NotValidUserException;

    void registration(@NonNull User user) throws UserAlreadyRegisteredException, NotValidUserException, MessagingException;

    void activateAccount(String code) throws NotFoundUserByActivationCode;


    void passwordRecovery(User user) throws NotValidEmailException, UserNotFoundException, MessagingException;
}

File: src/main/kotlin/com/example/service/AuthService.kt
package com.example.service

import com.example.utils.dto.UserWithoutLoginOrEmail
import com.example.utils.dto.responses.JwtResponse
import com.example.utils.dto.entities.User
import org.springframework.stereotype.Service

@Service
interface AuthService {

    fun registration(user: User)

    fun login(userWithoutLoginOrEmail: UserWithoutLoginOrEmail): JwtResponse

    fun activateAccount(code: String)

    fun passwordRecovery(userWithoutLoginOrEmail: UserWithoutLoginOrEmail)
}
